osgi/src/main/java/org/pentaho/ctools/cda/endpoints/CdaJettyWebsocket.java

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright 2018 Webdetails, a Hitachi Vantara company. All rights reserved.
+ *
+ * This software was developed by Webdetails and is provided under the terms
+ * of the Mozilla Public License, Version 2.0, or any later version. You may not use
+ * this file except in compliance with the license. If you need a copy of the license,
+ * please go to  http://mozilla.org/MPL/2.0/. The Initial Developer is Webdetails.
+ *
+ * Software distributed under the Mozilla Public License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. Please refer to
+ * the license for the specific language governing your rights and limitations.
+ */
+package org.pentaho.ctools.cda.endpoints;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.eclipse.jetty.websocket.WebSocket;
+import pt.webdetails.cda.push.IWebsocketEndpoint;
+
+import javax.servlet.http.HttpServletRequest;
+
+public class CdaJettyWebsocket implements WebSocket.OnTextMessage {
+  private static final Log logger = LogFactory.getLog( CdaJettyWebsocket.class );
+
+  private Connection connection;
+
+  private final HttpServletRequest request;
+  private final IWebsocketEndpoint websocketJsonQueryEndpoint;
+
+  CdaJettyWebsocket( HttpServletRequest request, IWebsocketEndpoint websocketJsonQueryEndpoint ) {
+    this.request = request;
+    this.websocketJsonQueryEndpoint = websocketJsonQueryEndpoint;
+  }
+
+  public void onOpen( Connection connection ) {
+    this.connection = connection;
+
+    this.websocketJsonQueryEndpoint.onOpen( this::processOutboundMessage );
```

Comment:
I would rather have the processOutboundMessage be a subscription of an event in the websocketJsonQueryEndpoint. This would allow for broadcast if we wished. We would have a bit more of state management on that class but I believe it would be better conceptually.
Nonetheless that is a change elsewhere and not directly on this PR.